bool    isValid
int32   counter
float32 offset
